### PR TITLE
Make ValueCreator accept structure for anon types instead name

### LINKED ISSUE #28525: Anonymous record creation fails in ValueCreator
**Description:**
The `ValueCreator` anonymous record creation is failing with the following issue.
```sh
		    Metrics publish failed: HTTP status code 499
		    	invalid content-type: text/plain
		    	MESSAGE DATA: Failed to dispatch inbound message. No such field or method: record {| string key; string value; |}
```

Here is the failed Github action(https://github.com/ballerina-platform/module-ballerina-grpc/pull/106/checks?check_run_id=1860698558)

**Steps to reproduce:**
Debug the gRPC module. Here is the debug information.

**Affected Versions:**
Swan Lake Alpha 3 SNAPSHOT

### PR DESCRIPTION
**Pull request by manuranga**
## Purpose
Make ValueCreator accept structure for anon types instead name

Fixes #28525

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRNode.java [hunk 1]
```diff
@@ -483,7 +490,7 @@ public String toString() {
 
         @Override
         public Name getName() {
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRNode.java:492]: Don't we need to have two functions to get name and displayName?

Comment (warunalakshitha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRNode.java:492]: Or this should be getDisplayName

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java [hunk 2]
```diff
@@ -505,28 +506,20 @@ void generateValueCreatorMethods(ClassWriter cw, List<BIRTypeDefinition> typeDef
...
-        }
-
-        i = 0;
-        for (BIRTypeDefinition optionalTypeDef : typeDefs) {
-            BType bType = optionalTypeDef.type;
-            if (bType.tag == TypeTags.OBJECT &&
-                    Symbols.isFlagOn(bType.tsymbol.flags, Flags.CLASS)) {
-                objectTypeDefs.add(i, optionalTypeDef);
-                i += 1;
+                recordTypeDefSet.add(optionalTypeDef);
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java:516]: AFAIU we have added a displayName to bir But I couldn't find that was used in codegen side. How that is working?

Comment (manuranga) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java:516]: displayName is used though `getName`

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRNode.java (hunk 1 above)
Comment (manuranga) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRNode.java:492]: `getName` is inherited though `NamedNode` so we can't change that name.

but we can rename `name` to `internalName` and `displayName` to `name`.

Comment (manuranga) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRNode.java]: fixed in b80de14

**Review by warunalakshitha — APPROVED** [review]

**Pull request merged**